src/cargo/core/compiler/mod.rs

```diff
@@ -913,6 +915,10 @@ fn build_base_args(
             .env("RUSTC_BOOTSTRAP", "1");
     }
 
+    if bcx.config.cli_unstable().future_incompat_report {
```

Comment:
This check doesn't get taken into account by Cargo's caching - adding/removing this flag from `cargo build` will not cause anything to rebuilt. I'm unsure how big of an issue this is, since all of this will go away once things get stabilized.